iris_markdown_EH.Rmd

```diff
@@ -0,0 +1,123 @@
+color2
+color2[species2 == "versicolor"] <- "blue"
+color2
+color2[species2 == "virginica"] <- "orange"
+color2
+```
+
+Cool, Now we can start doing some real analyses. 
+I have have made a best fit line of regression for Sepel Width as a function of Petal Length and have displayed the output stats:
+
+```{r}
+sepalW.lm <- lm(sepalW ~ petalL)
+sepalW.lm
+```
+
+
+This is a scatter plot for "Petal Length vs Petal Width" with a line of best fit.
+```{r}
+plot(petalL, sepalW, col=color2, cex=1.5, pch=19, xlab="Petal Length", ylab="Sepal Width")
+abline(sepalW.lm, col="black")
+legend('top', legend = levels(species), col = c("purple", "blue", "orange"), cex = .8, pch = 19)
+```
+
+As we can see from the graph above, the "general" trend is that as petal length increases, the sepal width decreases.
+
+BUT WAIT THERE'S MORE!!!!
+
+
+We can also index the iris dataset by species and make a regression line for Sepel Width as a function of Petal Length FOR EACH SPECIES and we can ALSO calcuate error for the linear regression FOR EACH SPECIES:
+```{r]
+setosa <- mix[mix$Species == "setosa",]
+setosa
+setosaPLvsSW.lm <- lm(setosa$Sepal.Width ~ setosa$Petal.Length)
+setosaPLvsSW.lm
+b.setosa <- coef(setosaPLvsSW.lm)
+int.setosa <- b.setosa[1]
+slope.setosa <- b.setosa[2]
+yhat.setosa <- slope.setosa*setosa$Petal.Length+int.setosa
+yhat.setosa
+error.setosa.sepalW <- setosa$Sepal.Width-yhat.setosa
+error.setosa.sepalW
+
+versicolor <- mix[mix$Species == "versicolor",]
+versicolor
+versicolorPLvsSW.lm <- lm(versicolor$Sepal.Width ~ versicolor$Petal.Length)
+versicolorPLvsSW.lm
+b.versicolor <- coef(versicolorPLvsSW.lm)
+int.versicolor <- b.versicolor[1]
+slope.versicolor <- b.versicolor[2]
+yhat.versicolor <- slope.versicolor*versicolor$Petal.Length+int.versicolor
+yhat.versicolor
+error.versicolor.sepalW <- versicolor$Sepal.Width-yhat.versicolor
+error.versicolor.sepalW
+
+virginica <- mix[mix$Species == "virginica",]
+virginica
+virginicaPLvsSW.lm <- lm(virginica$Sepal.Width ~ virginica$Petal.Length)
+virginicaPLvsSW.lm
+b.virginica <- coef(virginicaPLvsSW.lm)
+int.virginica <- b.virginica[1]
+slope.virginica <- b.virginica[2]
+yhat.virginica <- slope.virginica*virginica$Petal.Length+int.virginica
+yhat.virginica
+error.virginica.sepalW <- virginica$Sepal.Width-yhat.virginica
+error.virginica.sepalW
+```
+
+THEN WE CAN PLOT THE DATA AGAIN WITH EACH SPECIES HAVING THEIR OWN LINE OF REGRESSION:
+```{r}
+plot(petalL, sepalW, col=color2, cex=1.5, pch=19, xlab="Petal Length", ylab="Sepal Width")
+abline(setosaPLvsSW.lm, col="purple")
+abline(versicolorPLvsSW.lm, col="blue")
+abline(virginicaPLvsSW.lm, col="orange")
+legend('topright', legend = levels(species), col = c("purple", "blue", "orange"), cex = 1, pch = 19)
+```
+
+What we can see from THIS plot is that, within species, as petal length increases, sepal width ALSO increases. WOW, who would have thought.
+
+And this will conclude my simple anaylsis of the iris dataset!
+
```

Comment:
EXCELLENT JOB!!!! 💯

Reply:
The markdown doesnʻt run by itself, even without the working directory commented out.  It does work when it is run through the script, "iris_simpleanalysis_EH.R", but it produces an error when it is run alone:

> require(rmarkdown)
Loading required package: rmarkdown
> require("rmarkdown")
> #Sys.setenv(RSTUDIO_PANDOC="C:/Program Files/RStudio/bin/pandoc")
> render("iris_markdown_EH.Rmd", "html_document")


processing file: iris_markdown_EH.Rmd
  |....                                                           |   7%
  |........                                                       |  13%
  |.............                                                  |  20%
  |.................                                              |  27%
  |.....................                                          |  33%
  |.........................                                      |  40%
  |.............................                                  |  47%
  |..................................                             |  53%
  |......................................                         |  60%
  |..........................................                     |  67%
  |..............................................                 |  73%
  |..................................................             |  80%
  |.......................................................        |  87%
  |...........................................................    |  93%
label: unnamed-chunk-7
Quitting from lines 113-118 (iris_markdown_EH.Rmd) 
Error in abline(setosaPLvsSW.lm, col = "purple") : 
  object 'setosaPLvsSW.lm' not found

Looks like you forgot to copy a line or two of code into your markdown, but itʻs in your script.  Please make sure you do proper error testing in future.  -M